pkg/controllers/data.go

```diff
@@ -82,7 +84,27 @@ func (d *DataController) Save(w http.ResponseWriter, r *http.Request) {
 	}
 	respondWithJson(w, http.StatusCreated, thing)
 	defer r.Body.Close()
+}
+
+func (d *DataController) DeleteByDeviceID(w http.ResponseWriter, r *http.Request) {
```

Comment:
Shouldn't this, and other functions in here, return `error` so they can be handled appropriately? None of these operations are guaranteed to succeed and the caller should be aware when they don't, shouldn't they?

Reply:
Hi @lcbm, as we talked in a call, the controller functions shouldn't return any error. This happens, because, the mux HandleFunc doesn't expect an error. For this reason, I'll be resolving this conversation.